auth/token.go

```diff
@@ -0,0 +1,51 @@
+package auth
+
+import (
+	"crypto/subtle"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/nano-interactive/go-utils"
+)
+
+const (
+	AuthTypeBearer = "bearer"
+)
+
+func ParseTokenFromHeader(header string) ([]byte, error) {
+	components := strings.SplitN(header, " ", 2)
```

Comment:
Try to optimize this allocation if you can

Reply:
WONTFIX: Allocations are inevitable, code comprehensibility is not. I sincerely doubt the performance benefit outweighs clarity in this case.